File: java-advent-of-code/src/test/java/advent/twenty_eighteen/Day17ReservoirResearchTest.java
package advent.twenty_eighteen;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class Day17ReservoirResearchTest {
    @Test
    public void checkReal() {
        Day17ReservoirResearch im = new Day17ReservoirResearch("/twenty_eighteen/Day17-ReservoirResearch-input.txt");

        assertEquals(31158, im.getPart1Answer().intValue());
        assertEquals(25419, im.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        Day17ReservoirResearch im = new Day17ReservoirResearch("/twenty_eighteen/Day17-ReservoirResearch-example#1.txt");

        assertEquals(57, im.getPart1Answer().intValue());
        assertEquals(29, im.getPart2Answer().intValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_eighteen/Day17ReservoirResearchTest.scala
package advent.twenty_eighteen

import org.scalatest.{FlatSpec, Matchers}

class Day17ReservoirResearchTest  extends FlatSpec with Matchers {
  "2018 Day 17 - Input" should "calculate answers" in {
    val dp = new Day17ReservoirResearch("/twenty_eighteen/Day17-ReservoirResearch-input.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day17ReservoirResearch("/twenty_eighteen/Day17-ReservoirResearch-example#1.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }
}
